### ISSUE TITLE
Symfonisk remote

### ISSUE THREAD

**Issue by krsk8** [sanjoyg/dirigera_platform#126]
Have you looked at adding support for the Symfonisk remote (for the Sonos/Smfonisk speakers)? The latest version have two shortcut buttons in addition to speaker control. I got the Somrig remote to work fine with your platform, and hope you could look into the Symfonisk remote also. Thanks!

**Comment by sanjoyg** [comment]
Sure... I could... would u enable debug mode and try pressing the button to see if hub_events are generated.

**Comment by krsk8** [comment]
Sure. Great! I attach the part from the debug report with the relevant time stamps. The name I have assigned to the remote is lydkontroll_kjokken, as you can see. 
[HA dirigera debug.txt](https://github.com/user-attachments/files/18289349/HA.dirigera.debug.txt)

I also see this debug report regarding the dirigera_platform, which occured before I started pressing the Symfonisk buttons:
"2025-01-02 10:15:45.816 WARNING (Thread-4) [custom_components.dirigera_platform] Failed to convert 2025-01-02T09:15:45.000Z to date/time..."

And there are other debug warnings regarding IKEA Outlets, from what I can see. Let me know if you need to see the full debug report (with other time stamps).

Thanks!

**Comment by sanjoyg** [comment]
Please repeat the process with the new release. The date/time errors should go away but the issue you reported wont be fixed. After updating do send the logs again please.

**Comment by krsk8** [comment]
Ok. Thanks. I have tried to repeat the debug process, and pressed all the buttons on the remote (play/pause -> back -> forward -> volume up -> volume down -> shortcut btn 1 -> shortcut btn 2). Here’s the excerpt from the debug file which i believe is relevant. 

[HAss Symfonisk debug log 2025-03-08.txt.txt](https://github.com/user-attachments/files/19143330/HAss.Symfonisk.debug.log.2025-03-08.txt.txt)

**Comment by HanYolo44** [comment]
Hi. The shortcut buttons actually work already. I have 4 symfonisk gen 2 remotes. You have to let the Ikea app update the firmware first (takes a while) and then remove it and add it back to the Ikea app and then reload your Matter/Dirigera integration in Home Assistant.  You can use the shortcut buttons as single press, double press or hold then release.

**Comment by krsk8** [comment]
Hi HanYolo44, and thanks for following up. I have tried your setup, but I can’t see the two shortcut buttons. I’m on firmware 1.0.35 (and 2.737 for the hub). Could you elaborate how you set this up? I try to create an automation, but i can’t choose between any buttons. I can select single, double and long press, as you say, but I can’t designate that event to any of the two buttons. It is just designated to the Symfonisk switch as such, and any action I try to set up does not work when i press any button. Am I missing something? Do you use a blueprint or something? Thanks!

**Comment by HanYolo44** [comment]
Hi there, happy to help. A few things to help you replicate what I have:

1. Yes my firmware versions match yours.
2. Do you have the matter integration/bridge on the dirigera hub enabled? I'm using matter to control everything on my dirigera hub.
3. When you go to your remote in HA under devices, do you see all the buttons? I've attached a screenshot to show what it looks like. I've renamed the buttons to make it easier when doing automations. At first, they all just say "button" and you have to click to see which it maps to and then rename.
4. Once you've confirmed all the buttons are there, try single presses, double and hold then release for the shortcuts. You'll see in the device settings beside the buttons if the action is working. It will say pressed once, or pressed twice or held etc.
5. I'm using an automation. No Blueprint or anything else. The automation is a little tricky, I'll explain below:
6. I've connected the speakers I want to control via the dirigera hub to each symfonisk remote, letting the dirigera hub map the volume, next/previous and play/pause buttons automatically.  I map the 6 shortcuts (single press, double press, hold/release for each button) using HA. 

Once you have verified the buttons all work and are being read by HA (bullets 3/4 above), you can set up the automation.

The automation for the single dot and two dot shortcuts (same for the somrig shortcut remote from Ikea) needs both a trigger and an "and/if" to work. Why? I don't know. I just experimented and found this works.

For single and two press triggers:  set the trigger to entity (state), find your button and set it as the entity, leave the attribute, from and to all empty.  Then add an "and if" condition, entity event, find your button again, then set attribute to "event type" and "state", "pressed once" or "pressed twice".  Then it will work! See attached screen shot.

For hold and release on the one and two dot shortcuts: A bit simpler. Set the trigger to entity (state), find your button and set it as the entity. Set attribute to "event type", "from" to "held down" and "to" to "release after being held down".  You don't need an and if condition here. Screenshot also attached.

Let me know if you need any more clarification.

<img width="332" alt="Image" src="https://github.com/user-attachments/assets/dacb95d0-bdd6-4903-9eba-459f28d30620" />

<img width="878" alt="Image" src="https://github.com/user-attachments/assets/3326382a-be36-4cac-bb07-50da618ad336" />

<img width="915" alt="Image" src="https://github.com/user-attachments/assets/7eebeafb-5ac7-4daa-8f7f-719bd28fdc8c" />

**Comment by krsk8** [comment]
Thanks! So something is off on my side. I have attached the same screens from my HASS Automation interface. As you can see I do not have the “1 dot”/“2 dot” part. (In Norwegian, sorry for that. I can change language if it does not make sense.) 

I am pretty sure I have enabled Matter on the Dirigera hub, but I did go back and forth on Matter earlier, and the IKEA Home Smart app does not give any indication of the status, as far as I can see. I don’t have any Matter services or units set up in Home Assistant. Instead I use Sanjoygs integration, which works fine for the other IKEA units.

I may have added the Symfonisk remote to the IKEA system after I enabled Matter on Dirigera, if that might be an issue?

I’m happy for any tips as to where I could fail check further. I could try a reset of my Dirigera hub (again).

[

![Image](https://github.com/user-attachments/assets/975d3fae-1074-42fe-8526-a9596cd86e48)
![Image](https://github.com/user-attachments/assets/09c73ac4-8ee0-4b29-b75c-06f45856e3d0)
![Image](https://github.com/user-attachments/assets/6273267f-9437-499c-8c5c-8dc9b318c2ca)

](url)

**Comment by HanYolo44** [comment]
Ah, I forgot to mention that I used the matter integration in home assistant to connect my Dirigera hub to HA as well as using the dirigera integration.  I found I couldn't get the dirigera integration to use the Symfonisk remotely properly so am using the matter integration instead for this remote.

So make sure matter is enabled in your Ikea app for the hub, then connect to the hub again in HA using the matter integration. See my screenshots.

<img width="1469" alt="Image" src="https://github.com/user-attachments/assets/387d7519-03ee-4e4c-ba2c-7af8b03f19be" />

<img width="841" alt="Image" src="https://github.com/user-attachments/assets/ce1b639d-d3e8-4405-b202-411a89c5fd39" />

**Comment by krsk8** [comment]
YES! Finally! Thank you so much. Just the right size "teaspoon" (or ELI5-style) in your explanation and examples to make it doable. Great! This is one of the last main milestones for me, integrating the IKEA equipment with my gen.1 Sonos players, to get my home automation in shape. Lovely! I wish you all the best!

**Comment by HanYolo44** [comment]
Excellent, glad it has helped!
